java/src/main/native/src/TableJni.cpp

```diff
@@ -444,78 +445,57 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_gdfPartition(
-    }
-    cudf::jni::gdf_column_wrapper output_agg_column(agg_column->size, agg_dtype, false);
-    gdf_context ctxt{0, GDF_SORT, 0, 0};
-    std::vector<gdf_column *> cols = n_output_table.get_gdf_columns();
-    switch (op) {
-      case GDF_COUNT:
-        JNI_GDF_TRY(env, NULL,
-                    gdf_group_by_count(n_group_by_table.num_columns(), n_group_by_table.begin(),
-                                       agg_column, nullptr, cols.data(), output_agg_column.get(),
-                                       &ctxt));
-        break;
-      case GDF_MAX:
-        JNI_GDF_TRY(env, NULL,
-                    gdf_group_by_max(n_group_by_table.num_columns(), n_group_by_table.begin(),
-                                     agg_column, nullptr, cols.data(), output_agg_column.get(),
-                                     &ctxt));
-        break;
-
-      case GDF_MIN:
-        JNI_GDF_TRY(env, NULL,
-                    gdf_group_by_min(n_group_by_table.num_columns(), n_group_by_table.begin(),
-                                     agg_column, nullptr, cols.data(), output_agg_column.get(),
-                                     &ctxt));
-        break;
-
-      case GDF_SUM:
-        JNI_GDF_TRY(env, NULL,
-                    gdf_group_by_sum(n_group_by_table.num_columns(), n_group_by_table.begin(),
-                                     agg_column, nullptr, cols.data(), output_agg_column.get(),
-                                     &ctxt));
-        break;
-
-      case GDF_AVG:
-        JNI_GDF_TRY(env, NULL,
-                    gdf_group_by_avg(n_group_by_table.num_columns(), n_group_by_table.begin(),
-                                     agg_column, nullptr, cols.data(), output_agg_column.get(),
-                                     &ctxt));
-        break;
+    for (int i = 0; i < n_values.size(); i++) {
+      n_values_cols.push_back(n_input_table->get_column(n_values[i]));
     }
 
-    cols.push_back(output_agg_column.get());
+    cudf::table const n_keys_table(n_keys_cols);
+    cudf::table const n_values_table(n_values_cols);
 
-    cudf::jni::native_jlongArray native_handles(env, reinterpret_cast<jlong *>(cols.data()),
-                                                cols.size());
-    n_output_table.get_native_handles_and_release();
-    output_agg_column.release();
+    std::vector<cudf::groupby::hash::operators> ops;
+    for (int i = 0; i < n_ops.size(); i++) {
+      ops.push_back(static_cast<cudf::groupby::hash::operators>(n_ops[i]));
+    }
 
-    return native_handles.get_jArray();
+    std::pair<cudf::table, cudf::table> result =
+        cudf::groupby::hash::groupby(n_keys_table, n_values_table, ops);
+
+    std::vector<gdf_column *> output_columns;
+    try {
+      for (auto col = result.first.begin(); col != result.first.end(); col++) {
+        output_columns.push_back(*col);
+      }
+      for (auto col = result.second.begin(); col != result.second.end(); col++) {
+        output_columns.push_back(*col);
+      }
```

Comment:
This can be simplified to:
```c++
    try {
      std::vector<gdf_column *> output_columns;
      output_columns.reserve(result.first.num_columns() + result.second.num_columns());
      output_columns.insert(output_columns.end(), result.first.begin(), result.first.end());
      output_columns.insert(output_columns.end(), result.second.begin(), result.second.end());
```
Although technically we don't need the output_columns vector at all.  The code could allocate the native_handles array and then loop over these as above but instead of copying into a std::vector it can copy into the native_handles array directly.